third_party/amd/lib/Dialect/TritonAMDGPU/IR/Dialect.cpp

```diff
@@ -447,6 +447,45 @@ LogicalResult ConcatOp::verify() {
   return success();
 }
 
+LogicalResult LocalLoadPackedTransposedOp::verify() {
+  auto srcTy = getSrc().getType();
+  auto dstTy = getType();
+  auto srcShape = srcTy.getShape();
+
+  auto dotEnc = cast<DotOperandEncodingAttr>(dstTy.getEncoding());
+  auto sharedEnc =
+      cast<triton::gpu::SwizzledSharedEncodingAttr>(srcTy.getEncoding());
```

Comment:
This is not guarnteed (from some autogenerated code using ODS etc.) so the verifier will crash if we have IR not matching this expectation. That should be avoided given the purpose of verifier is to reject invalid IRs. We need to use `dyn_cast` and error out if these layout expections are unmet.